### ISSUE TITLE
Halo warning asks users for more halo points than required

### ISSUE THREAD

**Issue by navidcy** [CliMA/Oceananigans.jl#2277]
Halo warning is conservative

https://github.com/CliMA/Oceananigans.jl/blob/ae67ae2d426045104c25957c259a54790a6bd1ba/src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_model.jl#L120

Should be replaced with

```julia
throw(ArgumentError("The grid halo $user_halo must be larger or equal than $required_halo."))
```

**Comment by glwagner** [comment]
💡 

It's also wrong for `Flat` right?

**Comment by navidcy** [comment]
I think now Flat dimensions are dealt OK, see, e.g.,

https://github.com/CliMA/Oceananigans.jl/blob/main/src/Grids/automatic_halo_sizing.jl

but still the warning should be `must be at least equal to $required_halo`, right?

https://github.com/CliMA/Oceananigans.jl/blob/40ac9d0efff7a0c877df652e3f9114fe53693bc2/src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_model.jl#L221

**Comment by glwagner** [comment]
For sure.

**Comment by navidcy** [comment]
fixed it over at #2786
